src/handleInteraction.test.ts

```diff
@@ -0,0 +1,389 @@
+			await expect(handleInteraction(interaction, mockConsole)).resolves.toBeUndefined();
+			expect(mockInteractionFollowUp).toHaveBeenCalled();
+			expect(mockInteractionFollowUp).toHaveBeenCalledWith({
+				content: 'yo object in secret',
+				ephemeral: true,
+			});
+		});
+
+		test('sends an ephemeral reply to check DMs', async () => {
+			await expect(handleInteraction(interaction, mockConsole)).resolves.toBeUndefined();
+			expect(mockInteractionReply).toHaveBeenCalled();
+			expect(mockInteractionReply).toHaveBeenCalledWith({
+				content: expect.stringContaining('Check your DMs') as string,
+				ephemeral: true,
+			});
+			expect(mockUserSend).toHaveBeenCalled();
+			expect(mockUserSend).toHaveBeenCalledWith('yo DMs');
+		});
+
+		test('edits the reply message to check DMs if the interaction was deferred', async () => {
+			interaction = { ...interaction, deferred: true } as unknown as CommandInteraction;
+			await expect(handleInteraction(interaction, mockConsole)).resolves.toBeUndefined();
+			expect(mockInteractionEditReply).toHaveBeenCalled();
+			expect(mockInteractionEditReply).toHaveBeenCalledWith(
+				expect.stringContaining('Check your DMs') as string
+			);
+			expect(mockUserSend).toHaveBeenCalled();
+			expect(mockUserSend).toHaveBeenCalledWith({ content: 'yo DMs object' });
+		});
+	});
+
+	test('does nothing if the sender is a bot', async () => {
+		const mockExecute = jest.fn();
+		const mockCommand: Command = {
+			name: 'global-guild-test',
+			description: 'lolcat',
+			requiresGuild: false,
+			execute: mockExecute,
+		};
+		mockAllCommands.set(mockCommand.name, mockCommand);
+
+		const interaction = {
+			commandName: mockCommand.name,
+			options: { data: [] },
+			client: {
+				user: { id: selfUid },
+			},
+			user: {
+				bot: true,
+				id: otherUid,
+			},
+			inCachedGuild: () => false,
+			member: null,
+			guild: null,
+			channel: { type: ChannelType.DM },
+		} as unknown as CommandInteraction;
+
+		await expect(handleInteraction(interaction, mockConsole)).resolves.toBeUndefined();
+		expect(mockExecute).not.toHaveBeenCalled();
+	});
+
+	test('does nothing if the sender is us', async () => {
+		const mockExecute = jest.fn();
+		const mockCommand: Command = {
+			name: 'global-guild-test',
+			description: 'lolcat',
+			requiresGuild: false,
+			execute: mockExecute,
+		};
+		mockAllCommands.set(mockCommand.name, mockCommand);
+
+		const interaction = {
+			commandName: mockCommand.name,
+			options: { data: [] },
+			client: {
+				user: { id: selfUid },
+			},
+			user: {
+				bot: false,
+				id: selfUid,
+			},
+			inCachedGuild: () => false,
+			member: null,
+			guild: null,
+			channel: { type: ChannelType.DM },
+		} as unknown as CommandInteraction;
+
+		await expect(handleInteraction(interaction, mockConsole)).resolves.toBeUndefined();
+		expect(mockExecute).not.toHaveBeenCalled();
+	});
+
+	test('does nothing if the command is not found', async () => {
```

Comment:
As far as I know, Discord validates command requests against the list we give them. But if that list ever becomes out of date (we forget to deploy commands, or the deployment failed, or a few other cases) then some graceful failure is a good idea.

Right now, a log message gets sent after every command invocation indicating who sent it and what they sent. If the command is not valid, nothing more gets printed. We could change that with a `console.warn` though.

Ideas?

Reply:
One thing Ze Kaiser did is that it would send errors to a particular channel on a guild if something was broken, as well as to admins. I don't think we need it to DM admins, but maybe implementing the error channel for this would be a good idea? I don't think we need to do it in this PR since it's already big enough, but what are your thoughts on that solution?

Reply:
Sounds good to me, as long as we don't need to hard-code the error channel here. Env vars for the win, lol

Reply:
I'll make a TODO on it